cumulusci/tasks/bulkdata/generate_and_load_data.py

```diff
@@ -80,22 +104,38 @@ def _init_options(self, kwargs):
         self.batch_size = int(self.options.get("batch_size", self.num_records))
         if self.batch_size <= 0:
             raise TaskOptionsError("Batch size should be greater than zero")
-        self.class_path = self.options.get("data_generation_task")
-        self.data_generation_task = import_global(self.class_path)
+        class_path = self.options.get("data_generation_task")
+        if class_path:
+            self.data_generation_task = import_global(class_path)
+        else:
+            raise TaskOptionsError("No data generation task specified")
+
+        self.debug_dir = self.options.get("debug_dir", None)
+        self.database_url = self.options.get("database_url")
 
-        if self.database_url and self.batch_size != self.num_records:
-            raise TaskOptionsError(
-                "You may not specify both `database_url` and `batch_size` options."
-            )
+        if self.database_url:
+            engine, metadata = self._setup_engine(self.database_url)
+            tables = metadata.tables
+
+            if len(list(tables)) and not self.options.get("replace_database"):
+                raise TaskOptionsError(
+                    f"Database {self.database_url} has tables "
+                    f"({list(tables)}) "
+                    "but `replace_database` was not specified"
+                )
 
     def _run_task(self):
         with TemporaryDirectory() as tempdir:
             for current_batch_size, index in generate_batches(
                 self.num_records, self.batch_size
             ):
+                self.logger.info(
+                    f"Generating a bunch, current_bunch_size={current_batch_size} "
```

Comment:
The terminology here switches between `bunch` and `batch` - do we want `batch` across the board?

Reply:
The Generate and load infrastructure has a mechanism for "batching" which is different than the batching used in the LoadData task. I've added a picture here:

https://salesforce.quip.com/Yh4fA1OA1hAj

One of the things to discuss is whether it is helpful to have two different words (e.g. bunch and batch or set and batch) or whether using batch at two different levels is okay.

Reply:
@prescod I thought you decided to use batch for both, but this still says bunch.

Reply:
@prescod oh never mind, that was from https://github.com/SFDO-Tooling/CumulusCI/pull/1435 which came after this.